service-report/src/main/kotlin/org/veupathdb/service/mblast/report/jobs/FormatExecutor.kt

```diff
@@ -0,0 +1,118 @@
+package org.veupathdb.service.mblast.report.jobs
+
+import com.fasterxml.jackson.databind.node.ObjectNode
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.ThreadContext
+import org.veupathdb.lib.blast.Blast
+import org.veupathdb.lib.blast.formatter.BlastFormatter
+import org.veupathdb.lib.compute.platform.job.JobContext
+import org.veupathdb.lib.compute.platform.job.JobExecutor
+import org.veupathdb.lib.compute.platform.job.JobResult
+import org.veupathdb.lib.jackson.Json
+import org.veupathdb.service.mblast.report.Const
+import org.veupathdb.service.mblast.report.ext.makeOutFileName
+import org.veupathdb.service.mblast.report.model.JobConfig
+import org.veupathdb.service.mblast.report.service.mblast.MBlastQuerySvc
+import java.util.zip.ZipEntry
+import java.util.zip.ZipOutputStream
+import kotlin.io.path.listDirectoryEntries
+import kotlin.io.path.name
+
+class FormatExecutor : JobExecutor {
+  private val logger = LogManager.getLogger(javaClass)
+
+  override fun execute(ctx: JobContext): JobResult {
+    ThreadContext.put(Const.THREAD_CONTEXT_JOB_ID, ctx.jobID.string)
+
+    try {
```

Comment:
Nit: Could use `CloseableThreadContext` here in a try with resources

Reply:
:+1: Didn't know that existed.